src/types/signed.rs

```diff
@@ -0,0 +1,55 @@
+use crate::types::{Address, Bytes, H256, U256};
+use serde::{Deserialize, Serialize};
+
+/// Struct representing signed data returned from `Accounts::sign` method.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct SignedData {
+    /// The original method that was signed.
+    pub message: String,
+    /// The keccak256 hash of the signed data.
+    #[serde(rename = "messageHash")]
+    pub message_hash: H256,
+    /// V value.
+    pub v: u8,
+    /// R value.
+    pub r: H256,
+    /// S value.
+    pub s: H256,
+    /// The signature bytes.
+    pub signature: Bytes,
+}
+
+/// Transaction data for signing. The `Accounts::sign_transaction` method will
+/// fill optional fields with sane defaults when they are ommited.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct TransactionParameters {
+    /// Transaction nonce (None for account transaction count)
+    pub nonce: Option<U256>,
+    /// To address
+    pub to: Option<Address>,
+    /// Supplied gas (None for sensible default)
+    pub gas: Option<U256>,
+    /// Gas price (None for sensible default)
+    pub gas_price: Option<U256>,
+    /// Transfered value (None for no transfer)
+    pub value: Option<U256>,
+    /// Data (None for empty data)
```

Comment:
Same here. I think it would be best to drop the `Option`.

Reply:
Sounds good. We can drop `Option` for `to`, `value`, and `data`. I had used `Option` originally to signal that it providing this transaction parameter was optional and when it was not provided some default would be used. For the aforementioned fields, the default also happens to be `Default::default()` so we can get rid of `Option` for them.

Reply:
I'd keep it for `to` as well. There is a difference between contract creation and sending to `0x0` address.

Reply:
Ah I see it in `ethereum_transaction`, it gets encoded differently if it's there or if its missing.